### ISSUE TITLE
swarm cluster create isn't idempotent

### ISSUE THREAD

**Issue by hagzag** [shelleg/swarm-poc#9]
When re-running the ansible-playbook playbooks/deploy-swarm-cluster.yml a second time on an existing cluster you get the following error:

```
TASK [docker-swarm-manager : Register Leader] **********************************
Wednesday 05 October 2016  23:32:15 +0300 (0:00:00.104)       0:01:32.399 *****
fatal: [shelleg-manager1]: FAILED! => {"changed": true, "cmd": "docker swarm init --advertise-addr 172.16.1.111", "delta": "0:00:00.025090", "end": "2016-10-05 20:32:15.431757", "failed": true, "rc": 1, "start": "2016-10-05 20:32:15.406667", "stderr": "Error response from daemon: This node is already part of a swarm. Use \"docker swarm leave\" to leave this swarm and join another one.", "stdout": "", "stdout_lines": [], "warnings": []} 
```

I guess we should do the following:
1. Check the swarm status by running: `docker info` if cluster is active and is manager you will get:
2. Get the token

Check if swarms available:

```
Swarm: active
 NodeID: 5co2q261ithbyg877vv2182k6
 Is Manager: true
```

In addition you can get the token with something like:

For workers:

`docker swarm join-token worker -q
SWMTKN-1-0261m4i917htkqeabushmnxrnjczt7dde06efac22mdey9usdo-7inmkf4fqpblcdqsqhhdipcnx`

For manager token (to add more managers - perhapps infra01 should become a manager too ...)

`docker swarm join-token manager -q
SWMTKN-1-0261m4i917htkqeabushmnxrnjczt7dde06efac22mdey9usdo-7fnq586m4zzavif2r8xco049o`

**Comment by hagzag** [comment]
@doronshai - any update on this ?

**Comment by hagzag** [comment]
@doronshai I Fixed on dev branch, I am in the middle of changing the swarm master/slave config considering I needed tls / security to work.

**Comment by doronshai** [comment]
just saw it now. apologize. 
What is the status of this issue? Is there something I need to do ?

**Comment by hagzag** [comment]
@doronshai no biggie ... still not close to perfect but getting there ... slowly slowly but surely 👍

**Comment by hackndoes** [comment]
Should be OK now. you added the relevant check for swarm availability before trying to recreate it. this introduced a new issue that you are trying to join a master that is already the creator (by the init process) so for now (as long as we are only using a single manager which is also the initiator) I removed the manager join command from it. this is now idempotant due to some more changes I made. we should how ever improve this role to be able to use multiple hosts as managers (we will need that for scalability and HA of the swarm) I am closing the issue for now.
